### PR TITLE
chore: Prepare v2.10 (#4350

### PR DESCRIPTION
**Pull request by JorTurFer**
### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Relates to https://github.com/kedacore/keda/issues/4018

### CODE REVIEW

**Review by zroubalik — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -85,15 +107,15 @@ Here is an overview of all new **experimental** features:
 - **AWS Cloudwatch Scaler:** Make `metricName` and `namespace` optional when using `expression` ([#4334](https://github.com/kedacore/keda/issues/4262))
 - **Azure Pipelines Scaler**: Add new parameter to limit the jobs returned ([#4324](https://github.com/kedacore/keda/issues/4324))
 - **Azure Queue Scaler**: Fix azure queue length ([#4002](https://github.com/kedacore/keda/issues/4002))
-- **Azure Service Bus Scaler:** Use correct auth flows with pod identity ([#4026](https://github.com/kedacore/keda/issues/4026)|[#4123](https://github.com/kedacore/keda/issues/4123))
-- **Azure Service Bus Scaler:**: Improve way clients are created to reduce amount of ARM requests ([#4262](https://github.com/kedacore/keda/issues/4262))
+- **Azure Service Bus Scaler**: Improve way clients are created to reduce amount of ARM requests ([#4262](https://github.com/kedacore/keda/issues/4262))
```
Comment (zroubalik) [inline CHANGELOG.md:110]: Could you please fix the link here

Comment (JorTurFer) [reply CHANGELOG.md:110]: omw

Comment (JorTurFer) [reply CHANGELOG.md:110]: what happens with the link?

**Review by tomkerkhove — APPROVED** [review]

**Review by zroubalik — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (tomkerkhove) [reply CHANGELOG.md:110]: I don't see a problem as well; I'm ok with resolving

**Pull request merged**